liblangutil/Exceptions.h

```diff
@@ -121,23 +131,24 @@ class Error: virtual public util::Exception
 		TypeError,
 		SyntaxError,
 		Warning,
-		Info
+		Info,
+		Unknown
 	};
 
 	enum class Severity
 	{
 		Error,
 		Warning,
-		Info
+		Info,
+		Unknown
 	};
```

Comment:
We don't need a new severity level in the enum. Instead the function converting from `string` to `Severity` can just return `optional<Severity>`.

Reply:
Greetings of the day @cameel,

I'm confused what we're trying to achieve here. 

1.  Do we need to **try** to parse string.
2.  or we need to check if we can find something in that enum which we can return or return std::nullopt

or something else ?

Thanks & Regards
@shikharvashistha

Reply:
The latter. It's fine to return `nullopt` if it's something that does not look like a valid severity level. If you parse it into an enum you can then use `switch`. Thanks to this we'll get an error if someone adds a new severity level and forgets to update this.

The parsing logic should be close to the definition of `Severity` rather than in `SourceReferenceFormatter`. We already had it in `Exceptions.h` (https://github.com/ethereum/solidity/pull/11606#discussion_r703078009) but then we decided to move it because it was used only in one test. But I think it'll be useful in more places in the future.

Reply:
Done ! 👍